nilearn/glm/tests/test_second_level.py

```diff
@@ -574,9 +574,39 @@ def test_non_parametric_inference_permutation_computation():
                                                      mask=mask, n_perm=100)
 
         assert get_data(neg_log_pvals_img).shape == shapes[0][:3]
+        raise Exception(np.unique(neg_log_pvals_img.get_fdata()))
         del func_img, FUNCFILE, neg_log_pvals_img, X, Y
 
 
+def test_non_parametric_inference_tfce():
```

Comment:
This test fails because the t-statistics are all -inf, 0, or inf, which `_calculate_tfce` will fail on. Is there a simple alternative to the current simulated data that won't produce infs?

Reply:
Sorry,  i haven't checked in detail how this test is generated, but there should not be inf values in tests. This is a bug in the test generation imho.

Reply:
If I create 4 images instead of duplicating one image 4 times, the problem goes away. I don't know if it's worth propagating that fix to the other tests, considering it probably slows things down.